internal/version/tool_latest_version_finder.go

```diff
@@ -0,0 +1,194 @@
+		return releases[i].GetCreatedAt().After(releases[j].GetCreatedAt().Time)
+	})
+
+	for i := 0; i < len(releases); i++ {
+		release := releases[i]
+		// filter out drafts, pre-releases and releases that do not belong to the tool
+		if release.GetDraft() || release.GetPrerelease() || s.skipTag(release.GetTagName(), toolName) {
+			continue
+		}
+
+		v, err := semver.NewVersion(s.versionFromTag(releases[i], toolName))
+
+		if err != nil {
+			return false, nil, err
+		}
+
+		if currentVersion.Compare(v) < 0 {
+			return true, &ReleaseInformation{
+				Version:     v.Original(),
+				PublishedAt: release.GetPublishedAt().Time,
+			}, nil
+		}
+	}
+	return false, nil, nil
+}
+
+func isAtLeast24HoursPast(t time.Time) bool {
+	return !t.IsZero() && time.Since(t) >= time.Hour*24
+}
+
+func isHomebrew(tool string) bool {
+	brewFormulaPath, err := homebrewFormulaPath(tool)
+	if err != nil {
+		return false
+	}
+
+	executablePath, err := executableCurrentPath()
+	if err != nil {
+		return false
+	}
+
+	return strings.HasPrefix(executablePath, brewFormulaPath)
+}
+
+func homebrewFormulaPath(tool string) (string, error) {
+	formula := tool
+	brewFormulaPathBytes, err := exec.Command("brew", "--prefix", "--installed", formula).Output()
+	if err != nil {
+		return "", err
+	}
+
+	brewFormulaPath := strings.TrimSpace(string(brewFormulaPathBytes))
+
+	brewFormulaPath, err = filepath.EvalSymlinks(brewFormulaPath)
+	if err != nil {
+		return "", err
+	}
+
+	return brewFormulaPath, nil
+}
+
+func executableCurrentPath() (string, error) {
+	executablePath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	executablePath, err = filepath.EvalSymlinks(executablePath)
+	if err != nil {
+		return "", err
+	}
+
+	return executablePath, nil
+}
+
+func (s *latestReleaseVersionFinder) HasNewVersionAvailable(version, tool string) (newVersionAvailable bool, newVersion string, err error) {
+	if version == "" {
+		return false, "", nil
+	}
+	svCurrentVersion, err := semver.NewVersion(version)
+	if err != nil {
+		return false, "", err
+	}
+
+	if svCurrentVersion.Prerelease() != "" { // ignoring prerelease for code changes against master
+		*svCurrentVersion, err = svCurrentVersion.SetPrerelease("")
+		if err != nil {
+			return false, "", err
+		}
+	}
+
+	newVersionAvailable, newV, err := s.searchLatestVersionPerTool(svCurrentVersion, tool)
+	if err != nil {
+		return false, "", err
+	}
+
+	if newVersionAvailable && (!isHomebrew(tool) || isAtLeast24HoursPast(newV.PublishedAt)) {
+		return newVersionAvailable, newV.Version, nil
+	}
+
+	return false, "", nil
+}
+
+func (s *latestReleaseVersionFinder) PrintNewVersionAvailable(w io.Writer, v, tool, bin string) error {
+	newVersionAvailable, latestVersion, err := s.HasNewVersionAvailable(v, tool)
+
+	if err != nil {
+		return err
+	}
+	if newVersionAvailable {
+		var upgradeInstructions string
+		if isHomebrew(tool) {
+			upgradeInstructions = fmt.Sprintf(`To upgrade, run "brew update && brew upgrade %s".`, bin)
```

Comment:
please confirm that atlascli will be no homebrew as well

Reply:
from the [scope](https://docs.google.com/document/d/1ysaJdxncIEBHSysek93Ibr4f1st_thj-8_iM0g738mM/edit#heading=h.my8l8kwcujpo) yes. but also double-checking w/ Andrea